pkg/cmd/repo/view/view.go

```diff
@@ -0,0 +1,211 @@
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/utils"
+	"github.com/spf13/cobra"
+)
+
+type ViewOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	RepoArg string
+	Web     bool
+}
+
+func NewCmdView(f *cmdutil.Factory, runF func(*ViewOptions) error) *cobra.Command {
+	opts := ViewOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		BaseRepo:   f.BaseRepo,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "view [<repository>]",
+		Short: "View a repository",
+		Long: `Display the description and the README of a GitHub repository.
+
+With no argument, the repository for the current directory is displayed.
+
+With '--web', open the repository in a web browser instead.`,
+		Args: cobra.MaximumNArgs(1),
+		RunE: func(c *cobra.Command, args []string) error {
+			if len(args) > 0 {
+				opts.RepoArg = args[0]
+			}
+			if runF != nil {
+				return runF(&opts)
+			}
+			return viewRun(&opts)
+		},
+	}
+
+	cmd.Flags().BoolVarP(&opts.Web, "web", "w", false, "Open a repository in the browser")
+
+	return cmd
+}
+
+func viewRun(opts *ViewOptions) error {
+	var toView ghrepo.Interface
+	if opts.RepoArg == "" {
+		var err error
+		toView, err = opts.BaseRepo()
+		if err != nil {
+			return err
+		}
+	} else {
+		if isURL(opts.RepoArg) {
+			parsedURL, err := url.Parse(opts.RepoArg)
+			if err != nil {
+				return fmt.Errorf("did not understand argument: %w", err)
+			}
+
+			toView, err = ghrepo.FromURL(parsedURL)
+			if err != nil {
+				return fmt.Errorf("did not understand argument: %w", err)
+			}
+		} else {
+			var err error
+			toView, err = ghrepo.FromFullName(opts.RepoArg)
+			if err != nil {
+				return fmt.Errorf("argument error: %w", err)
+			}
+		}
+	}
+
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+
+	apiClient := api.NewClientFromHTTP(httpClient)
+
+	repo, err := api.GitHubRepo(apiClient, toView)
+	if err != nil {
+		return err
+	}
+
+	openURL := generateRepoURL(toView, "")
+	if opts.Web {
+		if opts.IO.IsStdoutTTY() {
+			fmt.Fprintf(opts.IO.ErrOut, "Opening %s in your browser.\n", displayURL(openURL))
+		}
+		return utils.OpenInBrowser(openURL)
+	}
+
+	fullName := ghrepo.FullName(toView)
+
+	readme, err := RepositoryReadme(apiClient, toView)
+	var notFound *api.NotFoundError
+	if err != nil && !errors.As(err, &notFound) {
+		return err
+	}
+
+	stdout := opts.IO.Out
+
+	if !opts.IO.IsStdoutTTY() {
+		fmt.Fprintf(stdout, "name:\t%s\n", fullName)
+		fmt.Fprintf(stdout, "description:\t%s\n", repo.Description)
+		if readme != nil {
+			fmt.Fprintln(stdout, "--")
+			fmt.Fprintf(stdout, readme.Content)
+			fmt.Fprintln(stdout)
+		}
+
+		return nil
+	}
+
+	repoTmpl := heredoc.Doc(`
+		{{.FullName}}
+		{{.Description}}
+		
+		{{.Readme}}
+		
+		{{.View}}
+		`)
```

Comment:
Nit: for consistency with how we style "heredocs" in struct literals, could this closing backtick be dedented one level?